sklearn/manifold/_isomap.py

```diff
@@ -124,11 +131,12 @@ class Isomap(TransformerMixin, BaseEstimator):
            framework for nonlinear dimensionality reduction. Science 290 (5500)
     """
     @_deprecate_positional_args
-    def __init__(self, *, n_neighbors=5, n_components=2, eigen_solver='auto',
-                 tol=0, max_iter=None, path_method='auto',
+    def __init__(self, *, n_neighbors=5, radius=None, n_components=2,
```

Comment:
Positional arguments are deprecated but still allowed, so I think we should add the new argument at the end to avoid breaking users' code.

Reply:
Hi @TomDLT , I've moved `radius` to the last argument as you suggested :)

Reply:
Thanks !
Additionally, it would be nice to have a small comment before `@_deprecate_positional_args` to say to future developers that once positional arguments are no longer accepted, we might want to move the parameter `radius` closer to the parameter `n_neighbors`.

Reply:
Hi @TomDLT , I've added the comments. I'm not sure if the phrasing is good enough, please feel free to make changes.

Reply:
Looks good, thanks !